tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2608,7 +2645,39 @@ private String handleDateFormatItem(String xpath, String value, boolean showCont
                 DateFormatSymbols dfs = sdf.getDateFormatSymbols();
                 dfs.setTimeSeparatorString(timeSeparator);
                 sdf.setDateFormatSymbols(dfs);
-                if (id == null || id.indexOf('B') < 0) {
+                if (parts.contains("availableFormats")) {
+                    String example = addExampleResult(sdf.format(DATE_SAMPLE) + ", " + sdf.format(DATE_SAMPLE5), "", showContexts);
+                    String altId = generateAltPattern(id);
+                    if (!altId.equals(id)) {
+                        String altXpath = "//ldml/dates/calendars/calendar[@type=\""
+                            + calendar
+                            + "\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\""
+                            + altId
+                            + "\"]";
+                        String forcedAltValue = cldrFile.getWinningValue(altXpath);
+                        String unforcedAltValue = generateAltPattern(value);
+                        String forcePrefix = forcedAltValue != null ? "" : EXAMPLE_OF_INCORRECT;
+                        
+                        
+                        sdf =
+                            icuServiceBuilder.getDateFormat(calendar, unforcedAltValue, numbersOverride);
+                        sdf.setTimeZone(ZONE_SAMPLE);
+                        defaultNumberingSystem =
+                                cldrFile.getWinningValue("//ldml/numbers/defaultNumberingSystem");
+                        timeSeparator =
+                                cldrFile.getWinningValue(
+                                        "//ldml/numbers/symbols[@numberSystem='"
+                                                + defaultNumberingSystem
+                                                + "']/timeSeparator");
+                        dfs = sdf.getDateFormatSymbols();
+                        dfs.setTimeSeparatorString(timeSeparator);
+                        sdf.setDateFormatSymbols(dfs);
+
+                        example = addExampleResult(forcePrefix + sdf.format(DATE_SAMPLE) + ", " + sdf.format(DATE_SAMPLE5), example, showContexts);
+                    }    
+                    return example;
```

Comment:
Should I account for the details checked below, e.g., `if (value.contains("MMM") || value.contains("LLL"))`?